packages/cli/src/modules/chat-hub/chat-hub-credentials.service.ts

```diff
@@ -73,4 +69,34 @@ export class ChatHubCredentialsService {
 
 		return credentials[PROVIDER_CREDENTIAL_TYPE_MAP[provider]]?.id ?? null;
 	}
+
+	async ensureWorkflowCredentials(
+		provider: ChatHubLLMProvider,
+		credentials: INodeCredentials,
+		workflowId: string,
+	) {
+		const credentialId = this.pickCredentialId(provider, credentials);
+		if (!credentialId) {
+			throw new BadRequestError('No credentials provided for the selected model provider');
+		}
+
+		const project = await this.sharedWorkflowRepository.getWorkflowOwningProject(workflowId);
+		if (!project) {
+			throw new ForbiddenError('Missing owner project for the workflow');
+		}
+
+		const allCredentials =
+			await this.credentialsService.findAllCredentialIdsForWorkflow(workflowId);
+
+		// If credential is shared through multiple projects just pick the first one.
```

Comment:
This comment confuses me a bit, because following lines only checks if credentialId is included in `allCredentials`